pkg/miscellaneous/miscellaneous.go

```diff
@@ -61,6 +61,59 @@ func GetLatestFullSnapshotAndDeltaSnapList(store brtypes.SnapStore) (*brtypes.Sn
 	return fullSnapshot, deltaSnapList, nil
 }
 
+type backup struct {
+	FullSnapshot      *brtypes.Snapshot
+	DeltaSnapshotList brtypes.SnapList
+}
+
+// GetFilteredSnapshots returns sorted by date (new -> old) SnapList. It will also filter the snapshots that should be included or not useing the filter funcion.
+// If the filter is nil it will return all snapshots. Also, maxNumberOfSnapshots can be used to target only the last N snapshots (0 = all).
+func GetFilteredSnapshots(store brtypes.SnapStore, maxNumberOfSnapshots int, filter func(snaps brtypes.Snapshot) bool) (brtypes.SnapList, error) {
```

Comment:
I am thinking that perhaps we should talk about "backups" and not "snapshots" here, see also my other comments.
```suggestion
func GetFilteredBackups(store brtypes.SnapStore, maxBackups int, filter func(snaps brtypes.Snapshot) bool) (brtypes.SnapList, error) {
```